src/reps/rep-utils.js

```diff
@@ -373,6 +376,25 @@ function getGripPreviewItems(grip) {
   return [];
 }
 
+function safeObjectLink(props, config, ...children) {
+  const {
+    objectLink,
+    object,
+  } = props;
+
+  if (objectLink) {
+    return objectLink(Object.assign({
+      object
+    }, config), ...children);
+  }
+
+  if (Object.keys(config).length === 0 && children.length === 1) {
+    return children[0];
+  }
+
+  return React.DOM.span(config, ...children);
+}
+
 module.exports = {
   createFactories,
```

Comment:
Yes, I agree.
I can still remove it here and file a bug on m-c to move it to a shared util and do the refactor where the one from reps is used for now.
How do you think we should operate ?